pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -264,13 +265,19 @@ protected void internalDeleteNamespace(AsyncResponse asyncResponse, boolean auth
         }
 
         boolean isEmpty;
-        List<String> topics;
+        Set<String> partitionedTopicNames = new HashSet<>();
+        Set<String> topicNames = new HashSet<>();
         try {
-            topics = pulsar().getNamespaceService().getListOfPersistentTopics(namespaceName)
-                    .get(config().getMetadataStoreOperationTimeoutSeconds(), TimeUnit.SECONDS);
-            topics.addAll(getPartitionedTopicList(TopicDomain.persistent));
-            topics.addAll(getPartitionedTopicList(TopicDomain.non_persistent));
-            isEmpty = topics.isEmpty();
+            if (policies == null || CollectionUtils.isEmpty(policies.replication_clusters)){
+                partitionedTopicNames.addAll(pulsar().getNamespaceService().getListOfPersistentTopics(namespaceName)
+                        .get(config().getMetadataStoreOperationTimeoutSeconds(), TimeUnit.SECONDS));
+            } else {
+                partitionedTopicNames.addAll(pulsar().getNamespaceService().getFullListOfTopics(namespaceName)
+                        .get(config().getMetadataStoreOperationTimeoutSeconds(), TimeUnit.SECONDS));
+            }
+            topicNames.addAll(getPartitionedTopicList(TopicDomain.persistent));
```

Comment:
`getFullListOfTopics()` includes the persistent and non-persistent topics.

Reply:
Yes.
`getFullListOfTopics()` includes the persistent and non-persistent topics, and `getPartitionedTopicList()` check the partitioned-topic by metadata., see also https://github.com/apache/pulsar/pull/17338#discussion_r960188195